fog/view/server/src/db_fetcher.rs

```diff
@@ -495,94 +525,151 @@ mod tests {
         assert_eq!(fetched_records.len(), blocks_and_records.len());
 
         for (i, fetched_record) in fetched_records.iter().enumerate() {
-            assert_eq!(fetched_record.ingest_invocation_id, invoc_id1);
-            assert_eq!(fetched_record.block_index, i as u64 + 10);
+            assert_eq!(fetched_record.ingress_key, ingress_key);
+            assert_eq!(fetched_record.block_index, i as u64 + 20);
             assert_eq!(blocks_and_records[i].1, fetched_record.records);
         }
 
         assert!(db_fetcher.get_pending_fetched_records().is_empty()); // The previous call should have drained this
 
+        let (highest_known_block_index, ingress_keys, missing_block_ranges) =
+            db_fetcher.get_highest_processed_block_context();
+        assert_eq!(
+            ingress_keys,
+            vec![IngressPublicKeyRecord {
+                key: ingress_key.clone(),
+                status: IngressPublicKeyStatus {
+                    start_block: 10,
+                    pubkey_expiry: 0,
+                    retired: false,
+                },
+                last_scanned_block: Some(29),
+            }]
+        );
+        assert_eq!(highest_known_block_index, 29);
+        assert!(missing_block_ranges.is_empty());
+
         // Add more blocks but this time leave a hole between the previous blocks and the new ones.
         // They should not get picked up untill a missed blocks range is reported.
-        let mut blocks_and_records = Vec::new();
-        for i in 30..40 {
+        let mut blocks_and_records_40_50 = Vec::new();
+        for i in 40..50 {
             let (block, records) = fog_test_infra::db_tests::random_block(&mut rng, i, 5); // 5 outputs per block
 
             db.add_block_data(&invoc_id1, &block, 0, &records).unwrap();
-            blocks_and_records.push((block, records));
+            blocks_and_records_40_50.push((block, records));
         }
 
         sleep(Duration::from_secs(1)); // Supposedly enough time for at least some blocks to get picked up.
 
         assert!(db_fetcher.get_pending_fetched_records().is_empty());
 
-        // Report a missing block range.
-        db.report_missed_block_range(&BlockRange::new(20, 30))
+        // TODO The last scanned block index gets updated even though we have a hole.
```

Comment:
@garbageslam I wonder what your thoughts on this are. There is a weird case where if somehow some blocks were skipped, `last_scanned_block` and `highest_known_block_index` would still go up. I think this is fine but wanted to check with you since this differs from the previous behavior, I think

Reply:
i think this is okay -- i think ingest only actually scans consecutive sequences.

if we wanted to change this, we could make the database enforce that consecutive sequences are scanned.

i don't think this is the wrong behavior for the view server to have in this scenario

Reply:
maybe its bad because, if there are gaps like this, reporting the key as lost won't fix it?

Reply:
idk, i can think of a few ways to try to fix this in the database if it actually happened, but i don't think it will actually happen because i think ingest asserts that it always processes consecutive blocks